datafusion/core/Cargo.toml

```diff
@@ -92,6 +92,8 @@ async-compression = { version = "0.4.0", features = [
     "tokio",
 ], optional = true }
 async-trait = { workspace = true }
+aws-config = "0.101.0"
+aws-credential-types = "0.101.0"
```

Comment:
Upgrade aws package to fix the audit failed message:
```
Crate:     rustls
Version:   0.20.9
error: 1 vulnerability found!
Title:     `rustls::ConnectionCommon::complete_io` could fall into an infinite loop based on network input
Date:      2024-04-19
ID:        RUSTSEC-2024-0336
URL:       https://rustsec.org/advisories/RUSTSEC-2024-0336
Severity:  7.5 (high)
Solution:  Upgrade to >=0.23.5 OR >=0.22.4, <0.23.0 OR >=0.21.[11](https://github.com/apache/datafusion/actions/runs/9321626823/job/25661037574?pr=10745#step:4:12), <0.22.0
Dependency tree:
rustls 0.20.9
├── tokio-rustls 0.23.4
│   └── hyper-rustls 0.23.2
│       └── aws-smithy-client 0.55.3
│           ├── aws-types 0.55.3
│           │   ├── aws-sig-auth 0.55.3
│           │   │   ├── aws-sdk-sts 0.28.0
│           │   │   │   └── aws-config 0.55.3
│           │   │   │       └── datafusion 38.0.0
│           │   │   │           ├── datafusion-wasmtest 38.0.0
│           │   │   │           ├── datafusion-substrait 38.0.0
│           │   │   │           ├── datafusion-sqllogictest 38.0.0
│           │   │   │           ├── datafusion-proto 38.0.0
│           │   │   │           │   └── datafusion-benchmarks 38.0.0
│           │   │   │           ├── datafusion-examples 38.0.0
│           │   │   │           ├── datafusion-docs-tests 38.0.0
│           │   │   │           └── datafusion-benchmarks 38.0.0
│           │   │   └── aws-sdk-sso 0.28.0
│           │   │       └── aws-config 0.55.3
│           │   ├── aws-sdk-sts 0.28.0
│           │   ├── aws-sdk-sso 0.28.0
│           │   ├── aws-http 0.55.3
│           │   │   ├── aws-sdk-sts 0.28.0
│           │   │   ├── aws-sdk-sso 0.28.0
│           │   │   └── aws-config 0.55.3
│           │   ├── aws-endpoint 0.55.3
│           │   │   ├── aws-sdk-sts 0.28.0
│           │   │   └── aws-sdk-sso 0.28.0
│           │   └── aws-config 0.55.3
│           ├── aws-sdk-sts 0.28.0
│           ├── aws-sdk-sso 0.28.0
│           └── aws-config 0.55.3
├── hyper-rustls 0.[23](https://github.com/apache/datafusion/actions/runs/9321626823/job/25661037574?pr=10745#step:4:24).2
└── aws-smithy-client 0.55.3
```
I'm not sure which version is better. Just pick up the latest version for version `0.xxx.xx`.